Feature/cmake rework without project options

StephanKa opened the pull request:
Completely reworked structure, moved more in direction of Jason Turner's boilerplate code without the project options at the moment.
- moved conan instantiation in conanfile.txt
- updated libraries
- added CMakePreset.json
- updated docker images
- updated workflow (make use of cmake presets)
- added stage for docker build testing
- updated README.md

StephanKa committed:
rework compiler in github action

StephanKa committed:
updated libraries, changed catch2 for being compatible with major version 3

StephanKa committed:
updated github actions, removed funding.yml

StephanKa committed:
changed name for standard, default is 20, added multiple cxx to pipeline

StephanKa committed:
changed CMake, updated gitignore, added gitattributes

- update pipeline, packages, fixed beast example
- updated readme and changed naming

StephanKa committed:
updated docker images

StephanKa committed:
Merge branch 'main' into feature/cmake-rework-without-project-options

arnemertz commented on .gitignore:
```diff
@@ -1,8 +1,39 @@
+CMakeUserPresets.json
+
+# IDE files
+.vs/
```
IDE and OS file ignores belong into each user's global .gitignore, not into the .gitignore of every single project.

StephanKa replied:
removed entries

arnemertz commented on README.md:
We might want to update the line about the origins, since now most of the CMake structure has changed to the boilerplate project instead of the GUI starter

StephanKa replied:
added hint

arnemertz commented on cmake/Options.cmake:
```diff
@@ -0,0 +1,40 @@
+OPTION(ENABLE_BUILD_WITH_TIME_TRACE "Enable -ftime-trace to generate time tracing .json files on clang" OFF)
```
shouldn't the compiler specific options be behind `IF(MSVC)` etc.?

StephanKa committed:
fixed review findings

- removed IDE and OS specifics from .gitignore
- added not MSVC check to ftrace cmake Options.cmake
- added hint for adaption to boilerplate repository by Jason Turner

StephanKa committed:
removed bincrafters conan remote
